### PR TITLE
feat: Add Python Blacken Docs

### PR DESCRIPTION
**Pull request by holtskinner**
## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [ ] I checked that all workflows return a success.
- [X] I included all the needed documentation for this change.
- [X] I provided the necessary tests.
- [X] I squashed all the commits into a single commit.
- [X] I followed the
      [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I wrote the necessary upgrade instructions in the
      [upgrade guide](../docs/upgrade-guide.md).
- [ ] If this pull request is about and existing issue, I added the
      `Fix #ISSUE_NUMBER` or `Close #ISSUE_NUMBER` text to the description of
      the pull request.

### Super-linter maintainer tasks

- [ ] Label as `breaking` if this change breaks compatibility with the previous
      released version.
- [ ] Label as either: `automation`, `bug`, `documentation`, `enhancement`,
      `infrastructure`.
- [ ] Add the pull request to a milestone, eventually creating one, that matches
      with the version that release-please proposes in the
      `preview-release-notes` CI job.

### CODE REVIEW

File: lib/globals/languages.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ LANGUAGE_ARRAY=('ANSIBLE' 'ARM' 'BASH' 'BASH_EXEC' 'CHECKOV' 'CLANG_FORMAT'
   'KUBERNETES_KUBECONFORM' 'KOTLIN' 'LATEX' 'LUA' 'MARKDOWN'
   "MARKDOWN_PRETTIER"
   'NATURAL_LANGUAGE' 'OPENAPI' 'PERL' 'PHP_BUILTIN' 'PHP_PHPCS' 'PHP_PHPSTAN'
-  'PHP_PSALM' 'POWERSHELL' 'PROTOBUF' 'PYTHON_BLACK' 'PYTHON_PYLINT'
+  'PHP_PSALM' 'POWERSHELL' 'PROTOBUF' 'PYTHON_BLACK' 'PYTHON_BLACKEN_DOCS' 'PYTHON_PYLINT'
```
Comment (ferrarimarco) [inline lib/globals/languages.sh:34]: Can you move `PYTHON_BLACKEN_DOCS` to a new line, and use double quotes? we're trying to move in that direction in this array.

File: scripts/linterVersions.sh [hunk 2]
```diff
@@ -54,6 +54,7 @@ LINTER_NAMES_ARRAY['PHP_PHPSTAN']="phpstan"
 LINTER_NAMES_ARRAY['PHP_PSALM']="psalm"
 LINTER_NAMES_ARRAY['PROTOBUF']="protolint"
 LINTER_NAMES_ARRAY['PYTHON_BLACK']="black"
+LINTER_NAMES_ARRAY['PYTHON_BLACKEN_DOCS']="blacken-docs"
```
Comment (ferrarimarco) [inline scripts/linterVersions.sh:57]: You need to add this to the `PATH`. I realized that we're missing this point from the instructions. Addressed in #6244

File: scripts/linterVersions.sh [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
```
Comment (ferrarimarco) [inline scripts/linterVersions.sh:1]: The updates to this file might be the result of an over-conservative rebase. Can you revert all the unrelated changes, and the duplicate lines that this change is trying to add?

File: scripts/linterVersions.sh [hunk 4]
```diff
@@ -231,7 +333,7 @@ for LANGUAGE in "${!LINTER_NAMES_ARRAY[@]}"; do
   elif [[ "${LINTER}" == "yamllint" ]]; then
     GET_VERSION_CMD="$("${LINTER}" --version | awk '{ print $2 }')"
   # Some linters don't support a "get version" command
-  elif [[ ${LINTER} == "bash-exec" ]] || [[ ${LINTER} == "gherkin-lint" ]]; then
+  elif [[ ${LINTER} == "bash-exec" ]] || [[ ${LINTER} == "blacken-docs" ]] || [[ ${LINTER} == "gherkin-lint" ]]; then
```
Comment (ferrarimarco) [inline scripts/linterVersions.sh:336]: I had a quick look at blacken-docs, and it doesn't appear to support an option to get its version. Can you please open an issue in their repository asking for that?

Thanks!

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Thanks for this PR.

I don't have experience with `blacken-docs`. From what I understand, it supports formatting Python code inside documentation files, and currently supports:

- Markdown
- reStructuredText
- LaTeX

Also, from blacken-docs readme:

> Additionally, you can run it on Python files to reformat Markdown and reStructuredText within docstrings.

It seems like this is the use case that we're trying to address here, because I don't see other file types besides Python files in the addition to `buildFileList.sh`

Do you know if this has any potential conflicts with Black?

Do you know if blacken-docs has evidence of being maintained? It looks like that it was just recently (~2 months ago) picked up after a 1 year pause, at least in terms of releases.

**Comment by ferrarimarco** [comment]
It seems that there are potential conflicts with Black, unfortunately. And blacken-docs doesn't appear to support a configuration file either.

https://github.com/adamchainz/blacken-docs/issues/62

**Comment by holtskinner** [comment]
> It seems that there are potential conflicts with Black, unfortunately. And blacken-docs doesn't appear to support a configuration file either.
> 
> [adamchainz/blacken-docs#62](https://github.com/adamchainz/blacken-docs/issues/62)

Ok, that's good to know... I suppose it doesn't make sense to add it into Superlinter at this time. I'll go ahead and close this out and remove blacken-docs from #6240

**Comment by ferrarimarco** [comment]
Thanks for the update, and sorry for the extra work!

**Pull request closed without merging**